diff --git a/config.def.h b/config.def.h
index 496d4f5dd6..ae37203398 100644
--- a/config.def.h
+++ b/config.def.h
@@ -1162,7 +1162,7 @@ static char buildbot_server_url[] = "http://buildbot.libretro.com/nightly/window
 static char buildbot_server_url[] = "http://buildbot.libretro.com/nightly/linux/x86_64/latest/";
 #elif defined(__i386__) || defined(__i486__) || defined(__i686__)
 static char buildbot_server_url[] = "http://buildbot.libretro.com/nightly/linux/x86/latest/";
-#elif defined(__arm__) && __ARM_ARCH == 7 && defined(__ARM_PCS_VFP)
+#elif defined(__arm__) && (__ARM_ARCH == 7 || __ARM_ARCH == 6) && defined(__ARM_PCS_VFP)
 static char buildbot_server_url[] = "http://buildbot.libretro.com/nightly/linux/armhf/latest/";
 #else
 static char buildbot_server_url[] = "";
